var/spack/repos/builtin/packages/proj/package.py

```diff
@@ -83,12 +84,45 @@ class Proj(AutotoolsPackage):
     )
 
     # https://proj.org/install.html#build-requirements
-    depends_on("pkgconfig@0.9.0:", type="build", when="@6:")
     depends_on("googletest", when="@6:")
     depends_on("sqlite@3.11:", when="@6:")
     depends_on("libtiff@4.0:", when="@7:+tiff")
     depends_on("curl@7.29.0:", when="@7:+curl")
+    depends_on("pkgconfig@0.9.0:", type="build", when="@6: build_system=autotools")
+    depends_on("cmake@2.6.0:", type="build", when="build_system=cmake")
 
+    build_system("autotools", conditional("cmake", when="@5.0.0:"), default="autotools")
```

Comment:
I've been setting the default to autotools incase the CMake build breaks something for some *nix user unexpectedly, and to just generally keep things for non Windows users fairly consistent so they don't have to think about Windows/related changes

Reply:
If we don't care about that I'll _happily_ change it (here and a number of other places I do the same thing).

Reply:
I would def prefer cmake over autotools, I've found cmake builds to be more reliable at dependency finding, and they also generate cmake-specific config files that get added to the installation prefix that help downstream packages that use them. Proj is built by our CI (torchgeo -> rasterio -> gdal -> proj) so should be heavily tested to make sure these kinds of changes don't break anything. But I would be fine for doing this with all packages in general, not just proj.

Reply:
Awesome. I may be a liiittle biased, but CMake is absolutely the way to go whenever possible, so I'm very on board with this approach. Thanks for supporting that!
I guess it would also only be changed in develop for now anyway, so on the off chance it did break things in some niche instance, that's not unreasonable.